src/poetry/core/packages/directory_dependency.py

```diff
@@ -101,9 +101,7 @@ def with_constraint(
             extras=list(self._extras),
         )
 
-        new._constraint = constraint
-        new._pretty_constraint = str(constraint)
-
+        new.set_constraint(constraint)
```

Comment:
before, this allowed `constraint` to be a `str`, although it seems the intention was for `set_constraint` to handle normalizing to `VersionConstraint` (as well as setting `_pretty_constraint`)

https://github.com/python-poetry/poetry-core/blob/f26e2361f0d643826d31366887dcb425fc078214/src/poetry/core/packages/dependency.py#L102